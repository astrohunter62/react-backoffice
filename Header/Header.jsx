import React, { Children } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { withStyles } from 'material-ui/styles'
import AppBar from 'material-ui/AppBar'
import Toolbar from 'material-ui/Toolbar'
import Typography from 'material-ui/Typography'
import IconButton from 'material-ui/IconButton'
import MenuIcon from 'material-ui-icons/Menu'

const drawerWidth = 280

const styles = (theme) => ({
  root: {
    width: '100%',
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: drawerWidth,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  hide: {
    display: 'none',
  },
  flex: {
    flex: 1,
  },
  title: {
    cursor: 'pointer',
    '&:hover': {
      textDecoration: 'underline',
    },
  }
})

const Header = ({
  title,
  open,
  fixed,
  handleDrawerOpen,
  onClick,
  children,
  classes
}) => (
  <div className={classes.root}>
    <AppBar
      className={classNames(classes.appBar, {
        [classes.appBarShift]: open,
      })}
      position={fixed ? 'fixed' : 'static'}
    >
      <Toolbar>
        <IconButton
          color="contrast"
          aria-label="open drawer"
          onClick={handleDrawerOpen}
          className={classNames(classes.menuButton, open && classes.hide)}
        >
          <MenuIcon />
        </IconButton>

        <Typography
          type="title"
          color="inherit"
          className={`${classes.flex} ${classes.title}`}
          onClick={onClick}
        >
          {title}
        </Typography>

        {children}
      </Toolbar>
    </AppBar>
  </div>
)

Header.propTypes = {
  title: PropTypes.string.isRequired,
  fixed: PropTypes.bool.isRequired,
  handleDrawerOpen: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Header)
