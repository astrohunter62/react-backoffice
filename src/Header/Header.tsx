import React, { FunctionComponent } from "react";
import classNames from "classnames";
import {
  AppBar,
  IconButton,
  Toolbar,
  Typography,
  withStyles,
} from "@material-ui/core";
import MenuIcon from "@material-ui/icons/Menu";

const drawerWidth = 280;

const styles = (theme: any) => ({
  appBar: {
    transition: theme.transitions.create(["margin", "width"], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    transform: "translate(0, 0)",
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(["margin", "width"], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: drawerWidth,
  },

  isStatic: {
    position: "relative",
  },

  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  hide: {
    display: "none",
  },
  flex: {
    flex: 1,
  },
  title: {
    cursor: "pointer",
    "&:hover": {
      textDecoration: "underline",
    },
  },
});

type HeaderProps = {
  isOpen?: boolean;
  title: string;
  isFixed?: boolean;
  onDrawerOpen: (...args: any[]) => any;
  onClick: (...args: any[]) => any;
  classes: {
    [key: string]: string;
  };
};

const Header: FunctionComponent<HeaderProps> = ({
  title,
  isOpen = false,
  isFixed = false,
  onDrawerOpen,
  onClick,
  children,
  classes,
}) => (
  <>
    <AppBar
      className={classNames(classes.appBar, {
        [classes.appBarShift]: isOpen,
        [classes.isStatic]: !isFixed,
      })}
      position={isFixed ? "fixed" : "static"}
    >
      <Toolbar>
        <IconButton
          aria-label="open drawer"
          color="inherit"
          onClick={onDrawerOpen}
          className={classNames(classes.menuButton, isOpen && classes.hide)}
        >
          <MenuIcon />
        </IconButton>

        <Typography
          variant="h6"
          color="inherit"
          className={`${classes.flex} ${classes.title}`}
          onClick={onClick}
        >
          {title}
        </Typography>

        {children}
      </Toolbar>
    </AppBar>
  </>
);

export default withStyles(styles as any)(Header);
