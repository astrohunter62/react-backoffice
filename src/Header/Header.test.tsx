import React from "react";
import { shallow } from "enzyme";

import Header from ".";

it("renders correctly", () => {
  const tree = shallow(
    <Header
      isOpen
      title="Header"
      isFixed
      isCookieInfoOpen
      onDrawerOpen={() => {}}
      onClick={() => {}}
    />,
  );

  expect(tree).toMatchSnapshot();
});

it("renders correctly with closed", () => {
  const tree = shallow(
    <Header
      title="Header"
      isFixed={false}
      isCookieInfoOpen={false}
      onDrawerOpen={() => {}}
      onClick={() => {}}
    />,
  );

  expect(tree).toMatchSnapshot();
});
