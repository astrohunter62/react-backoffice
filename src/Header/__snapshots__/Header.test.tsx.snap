// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Header
  classes={
    Object {
      "appBar": "Header-appBar-1",
      "appBarShift": "Header-appBarShift-2",
      "flex": "Header-flex-6",
      "hide": "Header-hide-5",
      "isStatic": "Header-isStatic-3",
      "menuButton": "Header-menuButton-4",
      "title": "Header-title-7",
    }
  }
  isFixed={true}
  isOpen={true}
  onClick={[Function]}
  onDrawerOpen={[Function]}
  title="Header"
/>
`;
